relm4-macros/src/factory/mod.rs

```diff
@@ -135,19 +135,23 @@ pub(crate) fn generate_tokens(
     };
 
     let last_segment = trait_.segments.last().unwrap();
-    let container_widget = match &last_segment.arguments {
-        syn::PathArguments::AngleBracketed(args) => {
-            args.args.first().map(|arg| arg.to_token_stream())
+
+    let (container_widget, parent_msg) = {
+        let mut args = match &last_segment.arguments {
+            syn::PathArguments::AngleBracketed(args) => Some(args.args.clone()),
+            _ => None,
         }
-        _ => None,
-    }
-    .unwrap_or_default();
+        .unwrap_or_default()
+        .into_iter();
+        (args.next().unwrap(), args.next().unwrap())
```

Comment:
Unwraps are probably a bad idea at this point, because macros are supposed to handle errors properly and you never know how many arguments are actually there. Maybe there is a good default for those values (the unit type maybe`) or we can avoid generics at this point entirely.

Reply:
I'm not sure in which kind of situation this would panic, but I understand your concern. For now, let's use `()` as the default.